rust/lanzatool/src/install.rs

```diff
@@ -8,37 +9,71 @@ use nix::unistd::sync;
 use tempfile::tempdir;
 
 use crate::esp::EspPaths;
-use crate::generation::Generation;
+use crate::gc::Roots;
+use crate::generation::{Generation, GenerationLink};
 use crate::pe;
 use crate::signature::KeyPair;
 
 pub struct Installer {
+    gc_roots: RefCell<Roots>,
     lanzaboote_stub: PathBuf,
     key_pair: KeyPair,
+    configuration_limit: u64,
     esp: PathBuf,
-    generations: Vec<PathBuf>,
+    generation_links: Vec<PathBuf>,
 }
 
 impl Installer {
     pub fn new(
         lanzaboote_stub: PathBuf,
         key_pair: KeyPair,
+        configuration_limit: u64,
         esp: PathBuf,
-        generations: Vec<PathBuf>,
+        generation_links: Vec<PathBuf>,
     ) -> Self {
         Self {
+            gc_roots: RefCell::new(Roots::new()),
             lanzaboote_stub,
             key_pair,
+            configuration_limit,
             esp,
-            generations,
+            generation_links,
         }
     }
 
     pub fn install(&self) -> Result<()> {
-        for toplevel in &self.generations {
-            let generation_result = Generation::from_toplevel(toplevel)
-                .with_context(|| format!("Failed to build generation from toplevel: {toplevel:?}"));
+        let mut links = self
+            .generation_links
+            .iter()
+            .map(GenerationLink::from_path)
+            .collect::<Result<Vec<GenerationLink>>>()?;
+
+        // A configuration limit of 0 means there is no limit
+        if self.configuration_limit > 0 {
+            // Sort the links by version
+            links.sort();
+
+            // Only install the number of generations configured
+            links = links
+                .into_iter()
+                .rev()
+                .take(self.configuration_limit as usize)
```

Comment:
nit: Technically this is a lossy conversion and that Rust allows this is sad. :) I tend to write `.try_into().unwrap()` when I don't explicitly intend a lossy conversion. But `configuration_limit` could just be `usize` to begin with.